File: 240116/키를 기준으로 정렬/sort-by-height.java
import java.util.Scanner;
import java.util.Arrays;
class human implements Comparable<human>{
    String name;
    int ki,age;

    public human(String name, int ki, int age){
        this.name = name;
        this.ki = ki;
        this.age = age;
    }

    public int compareTo(human human1){
        return this.ki - human1.ki;
    }
}
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        // 여기에 코드를 작성해주세요.
        
        int n = sc.nextInt();
        human[] people = new human[n];
        for(int i = 0 ; i<n; i++){
            String name = sc.next();
            int ki = sc.nextInt();
            int age = sc.nextInt();

            people[i] = new human(name,ki,age);
        }
        Arrays.sort(people);

        for(int i = 0 ; i<n; i++){
            System.out.println(people[i].name+" "+people[i].ki+" "+people[i].age);
        }

        
    }
}

File: 240116/키를 기준으로 정렬/sort-by-height.py
n = int(input())

class Student:

    def __init__(self,name,ki,age):
        self.name = name
        self.ki = ki 
        self.age = age

people = [] 

for _ in range(n):
    name , ki , age = map(str,input().split())

    people.append(Student(name,ki,age))

people.sort(key = lambda x:x.ki)

for i in range(n):
    print(people[i].name,people[i].ki,people[i].age)